pkg/cmd/server/kubernetes/node.go

```diff
@@ -144,42 +142,12 @@ func (c *NodeConfig) RunKubelet() {
 	}
 
 	cadvisorInterface, err := cadvisor.New(4194)
-	if err == nil {
-		// TODO: use VersionInfo after the next rebase
-		_, err = cadvisorInterface.MachineInfo()
-	}
 	if err != nil {
-		glog.Errorf("WARNING: cAdvisor cannot be started: %v", err)
-		cadvisorInterface = &cadvisor.Fake{}
+		glog.Fatalf("Error instantiating cadvisor: %v", err)
```

Comment:
It works fine in boot2docker as long as I build the binary with the release container (which I guess is the official way of building it anyway). It's just slower to build than a native cross-compile.